### PR TITLE
[#1] [CHORE] Bootstrap project by using react-templates

### LINKED ISSUE #1: [CHORE] Bootstrap project by using react-templates
## Why
To get started, we bootstrap our project by using [react-templates](https://github.com/nimblehq/react-templates)

## Acceptance Criteria
- Setup a new project by using the command from the above template and run the project successfully

### PR DESCRIPTION
**Pull request by manh-t**
Close #1 

## What happened 👀

- Bootstrap a new React application by using the [react-templates](https://github.com/nimblehq/react-templates)
- Add CODEOWNERS file
- Update all dependencies to latest version.

## Insight 📝

- Remove the CICD workflows for now.
- When updating the dependencies' versions, there is a problem with Jest:
```
Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /Users/imrankhan/Development/roc/node_modules/axios/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import axios from './lib/axios.js';
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

    > 1 | import axios, { AxiosRequestConfig } from 'axios';
        | ^
      2 |
      3 | class HttpClientService {
      4 |       /**

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1728:14)
      at Object.<anonymous> (src/app/services/http-client/HttpClient.service.tsx:1:1)
```
=> Fixed by adding `--transformIgnorePatterns` to `"test": "react-scripts test --transformIgnorePatterns",`

## Proof Of Work 📹

App runs successfully via `npm start`:
image

All tests passed:
```
 PASS  src/lib/requestManager.test.ts
 PASS  src/screens/Home/index.test.tsx

Test Suites: 2 passed, 2 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        1.135 s
Ran all test suites.

Watch Usage: Press w to show more.
```

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Team
+* @manh-t
```
Comment (nvminhtue) [inline .github/CODEOWNERS]: Could you add Rajib and me to the CODEOWNERS file?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# name: Deploy to Netlify
```
Comment (nvminhtue) [inline .github/workflows/deploy.yml]: Do we have any chores for applying those workflows? Or you decided you do it manually?

Update: I saw you updated it on later PRs. Do you think it's better to `remove` those unused workflows and freshly add in later PRs?

File: src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <App />
+      </BrowserRouter>
+    </Suspense>
+  </React.StrictMode>
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
Comment (nvminhtue) [inline src/index.tsx]: I don't think we need this. What do you think about getting rid of it 🤔

**Review by nvminhtue — COMMENTED** [review]
Only some minor concerns 👏

File: package.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "devDependencies": {
+    "@cypress/code-coverage": "^3.9.12",
+    "@cypress/instrument-cra": "^1.4.0",
+    "@nimblehq/eslint-config-nimble-react": "^1.2.0",
```
Comment (nvminhtue) [inline package.json]: let's add `@nimblehq/eslint-config-nimble-testing` for testing purpose linter as well
image

FYI: https://github.com/nimblehq/eslint-config-nimble/tree/develop/packages/eslint-config-nimble-react

File: .tool-versions [hunk 5]
```diff
@@ -0,0 +1 @@
+nodejs 16.14.2
```
Comment (nvminhtue) [inline .tool-versions]: Let's up to LTS 18.16.0

Comment (manh-t) [reply .tool-versions]: @nvminhtue cool, will update now 👍

File: .github/CODEOWNERS (hunk 1 above)
Comment (manh-t) [reply .github/CODEOWNERS]: @nvminhtue added in [b68ff78](https://github.com/manh-t/react-survey/pull/39/commits/b68ff7877896343f127df50578d91e3d4360f9f1)

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (manh-t) [reply .github/workflows/deploy.yml]: @nvminhtue removed in [b68ff78](https://github.com/manh-t/react-survey/pull/39/commits/b68ff7877896343f127df50578d91e3d4360f9f1)

File: src/index.tsx (hunk 3 above)
Comment (manh-t) [reply src/index.tsx]: @nvminhtue removed in [b68ff78](https://github.com/manh-t/react-survey/pull/39/commits/b68ff7877896343f127df50578d91e3d4360f9f1)

File: package.json (hunk 4 above)
Comment (manh-t) [reply package.json]: @nvminhtue added in [b68ff78](https://github.com/manh-t/react-survey/pull/39/commits/b68ff7877896343f127df50578d91e3d4360f9f1)

File: .tool-versions (hunk 5 above)
Comment (manh-t) [reply .tool-versions]: @nvminhtue added in [b68ff78](https://github.com/manh-t/react-survey/pull/39/commits/b68ff7877896343f127df50578d91e3d4360f9f1)

**Comment by manh-t** [comment]
@nvminhtue I resolved your comments and also upgraded the dependencies to latest versions 🙏

File: package.json [hunk 6]
```diff
@@ -3,22 +3,22 @@
...
+    "i18next-http-backend": "^2.2.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-i18next": "^11.16.1",
+    "react-i18next": "^12.3.1",
     "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
     "sass": "^1.49.11",
-    "web-vitals": "^2.1.4"
+    "web-vitals": "^3.3.2"
```
Comment (nvminhtue) [inline package.json]: I think we don't need it as removed the `reportWebVitals` entirely, right?

**Review by nvminhtue — APPROVED** [review]
LGTM already, good job @manh-t 👏

File: package.json (hunk 6 above)
Comment (manh-t) [reply package.json]: Removed ✅

File: .prettierrc [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+	"semi": true,
+	"singleQuote": true,
+	"trailingComma": "es5",
+	"printWidth": 130,
+	"tabWidth": 2
```
Comment (tyrro) [inline .prettierrc]: could we keep it to 2 spaces to remain consistent with other files?

Could you run all the files through `prettier`? This way we'll know whether the configuration file is working or not

```sh
npx prettier --write .
```

File: .prettierrc
Comment (tyrro) [inline .prettierrc:1]: could we also have a `.prettierignore`, as we have `stylelint`, `prettier` shouldn't format the `css` files

File: package.json [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "devDependencies": {
+    "@babel/plugin-proposal-private-property-in-object": "^7.21.0-placeholder-for-preset-env.2",
+    "@cypress/browserify-preprocessor": "3.0.2",
+    "@cypress/code-coverage": "^3.9.12",
```
Comment (tyrro) [inline package.json]: could we use the exact version? The idea is: using an exact version prevents unwanted minor/patch updates. 

```suggestion
    "@cypress/code-coverage": "3.9.12",
```

File: package.json [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    "stylelint": "stylelint '**/*.scss'",
+    "stylelint:fix": "stylelint '**/*.scss' --fix",
+    "codebase:lint": "npm run lint && npm run stylelint",
+    "codebase:fix": "npm run lint:fix && npm run stylelint:fix",
+    "cypress": "start-server-and-test start 3000 cypress:run",
+    "cypress:run": "cypress run",
+    "cypress:open": "cypress open"
+  },
+  "eslintConfig": {
+    "extends": "react-app"
```
Comment (tyrro) [inline package.json]: we don't need it as we have a `.eslintrc` file

File: stylelint.config.js
Comment (tyrro) [inline stylelint.config.js:1]: could we rename the file to `.stylelintrc ` to remain in sync with `.eslintrc` and `.prettierrc`?

**Review by tyrro — CHANGES_REQUESTED** [review]

File: .prettierrc (hunk 7 above)
Comment (manh-t) [reply .prettierrc]: @tyrro thank you, I add `.prettierignore` and run the `prettier` command 😄 . The update is in [e5c5101](https://github.com/manh-t/react-survey/pull/39/commits/e5c5101285ed5f2dd852132b3fd130bd9dddf7dc)

File: package.json (hunk 8 above)
Comment (manh-t) [reply package.json]: Updated in [e5c5101](https://github.com/manh-t/react-survey/pull/39/commits/e5c5101285ed5f2dd852132b3fd130bd9dddf7dc)

File: package.json (hunk 9 above)
Comment (manh-t) [reply package.json]: Updated in [e5c5101](https://github.com/manh-t/react-survey/pull/39/commits/e5c5101285ed5f2dd852132b3fd130bd9dddf7dc)

File: stylelint.config.js
Comment (manh-t) [reply stylelint.config.js:1]: Updated in [e5c5101](https://github.com/manh-t/react-survey/pull/39/commits/e5c5101285ed5f2dd852132b3fd130bd9dddf7dc)

File: .prettierignore [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+**/.svn
+**/.hg
+**/node_modules
+
+# Ignore artifacts:
+build
+coverage
+
+**/*.css
+**/*.md
```
Comment (tyrro) [inline .prettierignore:12]: I think `*.css` will match any files ending with `.css`, also please don't forget to update `*.css` to `*.scss` if you decide to use `sass` later

```suggestion
*.css
*.md
```

File: package.json [hunk 11]
```diff
@@ -0,0 +1,91 @@
+{
+  "name": "manh-react-survey",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "axios": "^1.4.0",
```
Comment (tyrro) [inline package.json]: please use the exact version for all the dependencies in this file

```suggestion
    "axios": "1.4.0",
```

File: package.json [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+  "scripts": {
+    "start": "react-scripts -r @cypress/instrument-cra start",
+    "build": "react-scripts build",
+    "test": "react-scripts test --transformIgnorePatterns",
+    "eject": "react-scripts eject",
+    "test:coverage": "rimraf coverage && react-scripts test --coverage --watchAll=false && npm run cypress:run && node ./scripts/coverage-merge.js && nyc report",
+    "test:merge-coverage": "node ./scripts/coverage-merge.js",
+    "lint": "eslint ./ --ext .js,.ts,.tsx",
+    "lint:fix": "eslint ./ --ext .js,.ts,.tsx --fix",
+    "stylelint": "stylelint '**/*.scss'",
```
Comment (tyrro) [inline package.json:22]: I see we are using `sass` here, could we rename all the files with `.css` extension to `.scss` and run `npx stylelint:fix`?

File: package.json [hunk 13]
```diff
@@ -0,0 +1,91 @@
+{
+  "name": "manh-react-survey",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "axios": "^1.4.0",
+    "i18next": "^22.5.1",
```
Comment (tyrro) [inline package.json]: Do we have a plan to use `i18n` in this project, if no, please remove this and the below two packages as well

**Review by tyrro — CHANGES_REQUESTED** [review]

File: .prettierignore (hunk 10 above)
Comment (manh-t) [reply .prettierignore:12]: Nice catch, will update it 🙏

File: package.json (hunk 12 above)
Comment (manh-t) [reply package.json:22]: sure 👍

File: package.json (hunk 13 above)
Comment (manh-t) [reply package.json]: @tyrro I think we don't use `i18n` in this IC. Let me remove it 🙏

**Comment by manh-t** [comment]
@tyrro All updated

File: .eslintignore [hunk 14]
```diff
@@ -0,0 +1,2 @@
+node_modules
+build
```
Comment (tyrro) [inline .eslintignore]: new line please

File: public/locales/en/translation.json
Comment (tyrro) [inline public/locales/en/translation.json:1]: we can delete this file now, right?

File: src/__mocks__/i18n.ts
Comment (tyrro) [inline src/__mocks__/i18n.ts:1]: this one as well  🙌🏿

**Review by tyrro — COMMENTED** [review]
there are some files related to i18n we don't need now probably, could we get rid of those?

File: .eslintignore (hunk 14 above)
Comment (manh-t) [reply .eslintignore]: @tyrro Removed in [f7908f4](https://github.com/manh-t/react-survey/pull/39/commits/f7908f44155a599551ebd98cc0f9b826151d6483)

File: public/locales/en/translation.json
Comment (manh-t) [reply public/locales/en/translation.json:1]: @tyrro Removed in [f7908f4](https://github.com/manh-t/react-survey/pull/39/commits/f7908f44155a599551ebd98cc0f9b826151d6483)

File: src/__mocks__/i18n.ts
Comment (manh-t) [reply src/__mocks__/i18n.ts:1]: @tyrro Removed in [f7908f4](https://github.com/manh-t/react-survey/pull/39/commits/f7908f44155a599551ebd98cc0f9b826151d6483)

File: package.json [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+  },
+  "jest": {
+    "collectCoverageFrom": [
+      "src/**/*.{js,jsx,ts,tsx}",
+      "!src/**/*.d.ts"
+    ],
+    "coverageReporters": [
+      "json"
+    ]
+  },
```
Comment (tyrro) [inline package.json]: `jest` recommends extracting the configurations into a separate file, it also keeps the `package.json` a bit cleaner

image

could we do that here

```js
// jest.config.js
/** @type {import('jest').Config} */
const config = {
  collectCoverageFrom: [
      "src/**/*.{js,jsx,ts,tsx}",
      "!src/**/*.d.ts"
    ],
    coverageReporters: [
      "json"
    ]
};

export default config;
```

**Review by tyrro — APPROVED** [review]
approved with a final comment about jest configuration file, great job 🎉

File: package.json (hunk 15 above)
Comment (manh-t) [reply package.json]: Thanks, I create `jest.config.ts` in [90dfded](https://github.com/manh-t/react-survey/pull/39/commits/90dfdedb959020ae7eaf841bb008a4f8a16dde6b) 🎉

**Pull request merged**